Source/WebKit/Scripts/generate-serializers.py

```diff
@@ -1261,15 +1367,179 @@ def parse_serialized_types(file):
+            if struct_or_class == 'webkit_secure_coding':
+                match = re.search(r'(.*): ([^;]*)', line)
+            else:
+                match = re.search(r'(.*) ([^;]*)', line)
             if match:
                 member_type, member_name = match.groups()
-                members.append(MemberVariable(member_type, member_name, member_condition, []))
+                if struct_or_class == 'webkit_secure_coding':
+                    dictionary_members.append(MemberVariable(member_type, member_name, member_condition, []))
+                else:
+                    members.append(MemberVariable(member_type, member_name, member_condition, []))
     return [serialized_types, serialized_enums, headers, using_statements, additional_forward_declarations]
 
 
+# When describing a webkit_secure_coding dictionary validation format, the MemberVariable's "type" is the dictionary string key,
+# and the MemberVariables "name" is the expected value type.
+def generate_one_dictionary_member_validation(member):
+    result = []
+    key = member.type
+    value = member.name
+    optional = value.endswith('?')
+    if optional:
+        value = value[:-1]
+
+    match = re.search(r'(.*)<(.*)>', value)
+    collection_key_type = None
+    collection_value_type = None
+    if match:
+        value, collection_value_type = match.groups()
+
+    if collection_value_type is not None:
+        match = re.search(r'(.*), (.*)', collection_value_type)
+        if match:
+            collection_key_type, collection_value_type = match.groups()
+
+    if optional:
+        result.append('    if (!dictionary.keyIsMissingOrHasValueOfType("' + key + '"_s, IPC::NSType::' + value + '))')
+    else:
+        result.append('    if (!dictionary.keyHasValueOfType("' + key + '"_s, IPC::NSType::' + value + '))')
+    result.append('        return false;')
+
+    if collection_value_type is not None:
+        if collection_key_type is not None:
+            result.append('    if (!dictionary.collectionValuesAreOfType("' + key + '"_s, IPC::NSType::' + collection_key_type + ', IPC::NSType::' + collection_value_type + '))')
+        else:
+            result.append('    if (!dictionary.collectionValuesAreOfType("' + key + '"_s, IPC::NSType::' + collection_value_type + '))')
+        result.append('        return false;')
+
+    result.append('')
+    return result
+
+
+def generate_webkit_secure_coding_impl(serialized_types):
+    serialized_types = resolve_inheritance(serialized_types)
+    result = []
+    result.append(_license_header)
+    result.append('#include "config.h"')
+    result.append('#include "GeneratedWebKitSecureCoding.h"')
+    result.append('')
+
+    header_set = set()
+    header_set.add(ConditionalHeader('"ArgumentCodersCocoa.h"', None))
+    for type in serialized_types:
+        if not type.is_webkit_secure_coding_type():
+            continue
+        if type.custom_secure_coding_header is not None:
```

Comment:
Then this isn't needed.  Just pass in a list of ConditionalHeaders for the secure coding .mm file.